packages/perennial-vaults/contracts/balanced/BalancedVaultDefinition.sol

```diff
@@ -62,19 +62,21 @@ contract BalancedVaultDefinition is IBalancedVaultDefinition {
     uint256 private immutable weight1;
 
     constructor(
-        Token18 asset_,
         IController controller_,
         UFixed18 targetLeverage_,
         UFixed18 maxCollateral_,
         MarketDefinition[] memory marketDefinitions_
     ) {
-        asset = asset_;
+        if (targetLeverage_.eq(UFixed18Lib.ZERO)) revert BalancedVaultDefinitionZeroTargetLeverageError();
```

Comment:
I'm wondering if it'd be good to actually have the ability to target 0 leverage as sort of a close-only setting for the vault

Reply:
I think that's reasonable, but this is similar to the fact that we don't allow all weights to be 0. Would we ever want to construct a vault with 0 targetLeverage?

Reply:
I'm in favor of having the check here and then if we ever do want to deploy a vault with 0 leverage (or 0 total weight) we have to explicitly remove the checks. This makes it safer in the normal case

Reply:
Gonna leave the check here, happy to discuss more though.